lib/chef/run_list/run_list_expansion.rb

```diff
@@ -131,6 +138,11 @@ def errors
         @missing_roles_with_including_role.map {|item| item.first }
       end
 
+      def parse_into_json
+        @seen_items = {:recipe => {}, :role => {}}
+        {:id => @environment, :run_list => convert_run_list_trace('top level')}.to_json
+      end
```

Comment:
Awesome, this is just what I was looking for. I had a feeling I should use JSONCompat but I couldn't find an example of how to do it with a complex hash building method as well. Thanks for the lead.